scripts/SREP/retry-pruning-cronjob/script.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+# Delete jobs that fail in openshift-sre-pruning namespace.
+
+set -x
+set -e
+set -o nounset
+set -o pipefail
+
+readonly PRUNING_NS=openshift-sre-pruning
+FailedJobs=()
+
+detect_job() {
+  echo "INFO: finding jobs producing error"
+  for current_ns in ${PRUNING_NS}; do
+    FailedJobs+=($(oc get jobs -n "$current_ns" -o=jsonpath='{.items[?(@.status.failed>0)].metadata.name}'))
+  done
+}
+
+delete_job() {
+  if [[ ${#FailedJobs[@]} == 0 ]]; then
+    echo "INFO: no failed jobs found exiting.."
+    return 1
+  else
+    echo "INFO: deleting jobs producing error"
+    for jobs in "${FailedJobs[@]}"; do
+      oc delete job "$jobs" -n "$PRUNING_NS"
+      FailedJobs=("${FailedJobs[@]:1}")
+    done
+  fi
+}
+
+retry_job() {
+  detect_job
```

Comment:
This function is redundant since it is doing the same thing as the main function. 

If it's our intention to retry (actually, it's just detect and delete) one more time, we just have the `retry_job` to be called twice in the main, may be just renaming the function to better reflect it's actual behavior? 
```
handle_failed_jobs() {
  detect_job
  delete_job
}

main() {
  handle_failed_jobs
  handle_failed_jobs
  echo "Succeed."
  exit 0
}
```

Reply:
Hi @xiaoyu74 , as discussed with meng bo, it was actually intended to create the failed job. I've done that but waiting for further reviews from all.